### ISSUE TITLE
MariaDB 10.5.15 breaks a test

### ISSUE THREAD

**Issue by WikiRik** [sequelize/sequelize#14174]
## Issue Creation Checklist

[X] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Issue Description

MariaDB 10.5.15 breaks [one test](https://github.com/sequelize/sequelize/blob/main/test/integration/transaction.test.js#L541-L660)

```
[MARIADB] Transaction
       deadlock handling
         should release the connection for a deadlocked transaction (2/2):

      AssertionError: expected false to be true
      + expected - actual

      -false
      +true
```

### Additional context

- [Dockerhub for MariaDB](https://hub.docker.com/_/mariadb)
- [Code for the Docker image](https://github.com/MariaDB/mariadb-docker/tree/master/10.5)

### Is this issue dialect-specific?

- [ ] No. This issue is relevant to Sequelize as a whole.
- [X] Yes. This issue only applies to the following dialect(s): MariaDB 10.5.15
- [ ] I don't know.

### Would you be willing to resolve this issue by submitting a Pull Request?

- [X] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by WikiRik** [comment]
Fix for the short term is included in #14171 by pinning the patch version of MariaDB. Renovate will do automatic updates on this so it is not an issue.

**Comment by WikiRik** [comment]
@ephys I only updated it for the main branch. This is also the failing test on v6. Renovate does not do automatic updates on v6 so we can either disable the test for now or fix it. But I'm not sure when I can get to fixing it

**Comment by ephys** [comment]
I made a branch to fix it a little while ago but changed my mind and opted to fix the test instead
Now that a v6 release is blocked by it I should prioritize it 😅

**Comment by WikiRik** [comment]
@ephys would be nice if you could take another look, otherwise I can check it out next week

**Comment by ephys** [comment]
The test fails because it tries to cause a deadlock but fails. MariaDB may have changed their heuristics to reduce the number of deadlocks:

Here is the query execution order: (ONE & TWO are the connection IDs of each transaction)

```
Executing (ONE): START TRANSACTION;
Executing (ONE): SELECT `id`, `username`, `awesome` FROM `users` AS `user` WHERE `user`.`id` = 1 LOCK IN SHARE MODE;
Executing (TWO): SET TRANSACTION ISOLATION LEVEL READ COMMITTED;
Executing (TWO): START TRANSACTION;
Executing (TWO): SELECT `id`, `username`, `awesome` FROM `users` AS `user` WHERE `user`.`id` = 1;
Executing (TWO): UPDATE `users` SET `awesome`=? WHERE `id` = ?
	 -> sets awesome to false

Executing (ONE): UPDATE `users` SET `awesome`=? WHERE `id` = ?
	-> sets awesome to true

Executing (ONE): COMMIT;
Executing (TWO): COMMIT;
```

final result: awesome is false

previously this lead to a deadlock instead

**Comment by ephys** [comment]
As a temporary workaround, we could disable this test for mariaDB 10.5.15. I'm starting to think this is an issue in MariaDB itself and it may get resolved in a future update?

**Comment by WikiRik** [comment]
Yeah, it might be better to skip this test with an expiring TODO for June 1st so we can take a look again if we want to release a stable build around that time. Can you make PRs for v6 and main for this or shall I do that?

**Comment by ephys** [comment]
This issue also happens with 10.7.3

**Comment by WikiRik** [comment]
Yeah, it could also be intended but then we look into it later. It's not the test with the highest impact luckily
